src/EcoSurveyComponent.java

```diff
@@ -0,0 +1,147 @@
+    private Map<Species, Integer> speciesCount;
+
+    private double area;
+
+    private int totalSpecies;
+
+    private void createNewRep() {
+        this.totalSpecies = 0;
+        this.area = 0;
+        this.speciesCount = new Map1L<>();
+    }
+
+    public EcoSurveyComponent() {
+        this.createNewRep();
+    }
+
+    //private method
+    private boolean hasSpecies(String name) {
+        //this function seems all wrong
+        boolean speciesFound = false;
+        for (Map.Pair<Species, Integer> s : this.speciesCount) {
+            if (s.key().name == name) {
+                speciesFound = true;
+            }
+        }
+        return speciesFound;
+    }
+
+    //I think this is a kernel function. I do want client to use it.
+    //Is it ok if it accesses fields/representation.
+    private void addSpecies(String name, String trueName, double density) {
+        assert !this.hasSpecies(name) : "Violation of species does not exist";
+
+        Species s1 = new Species();
+        s1.name = name;
+        s1.trueName = trueName;
+        s1.density = density;
+        this.speciesCount.add(s1, 0);
+        this.totalSpecies++;
+    }
+
+    //kernel. Should these methods be public?
+    private void setCount(String name, int count) {
+        assert this.hasSpecies(name) : "Violation of species exists";
+
+        Species s1 = this.getSpecies(name);
+        this.speciesCount.replaceValue(s1, count);
+
+    }
+
+    private Species getSpecies(String name) {
+        assert this.hasSpecies(name) : "Violation of species exists";
+
+        Species s1 = new Species();
+        for (Map.Pair<Species, Integer> s : this.speciesCount) {
+            if (s.key().name == name) {
+                //aliasing? May need to make a instance methods
+                s1 = s.key();
+            }
+        }
+        return s1;
+    }
+
+    private void setCoordinates(double a, double b, double c, double d) {
+        //double four = 4d;
+        this.area = (a + b + c + d) / 4;
+        //
+        /**
+         * obviously this formula is not good enough. I want to calculate the
+         * area of the survey plot in this function. Not sure if I need to track
+         * the coordinates afterwards
+         *
+         */
+    }
+
+    public String getStatus(String name) {
+        Species s1 = this.getSpecies(name);
+        int count = this.speciesCount.value(s1);
+        double density = s1.density;
+        String status = "";
+        if (count / this.area < density) {
+            status = "Critical";
+        } else if (count / this.area > 2 * density) {
+            status = "Thriving";
+        } else {
+            status = "Normal";
+        }
+        return status;
+    }
+
+    public static void main(String[] args) {
+        SimpleReader in = new SimpleReader1L();
+        SimpleWriter out = new SimpleWriter1L();
+
+        //This made me realize I need a better way to iterate through
```

Comment:
You can also make your object implement iterable, if you want. Otherwise, the user needs a removeAny style method for iteration.